src/lsm/groove.zig

```diff
@@ -577,37 +576,59 @@ pub fn GrooveType(
 
             groove.prefetch_snapshot = snapshot_target;
             assert(groove.prefetch_objects.count() == 0);
-            assert(groove.prefetch_ids.count() == 0);
+            assert(groove.prefetch_keys.count() == 0);
         }
 
         /// This must be called by the state machine for every key to be prefetched.
         /// We tolerate duplicate IDs enqueued by the state machine.
         /// For example, if all unique operations require the same two dependencies.
         pub fn prefetch_enqueue(groove: *Groove, key: PrimaryKey) void {
-            if (!has_id) {
-                groove.prefetch_ids.putAssumeCapacity(key, {});
-                return;
+            if (has_id) {
+                if (!groove.ids.key_range_contains(key)) return;
```

Comment:
I think we can only use `groove.ids.key_range_contains` when we are querying `snapshot_latest`.

Reply:
Let's pass in the lookup-snapshot to `key_range_contains`, the same way that `ManifestLevel` handles this.

Reply:
Mhmmmmm, great catch! I had somehow (wrongly) convinced myself that the shape of the tree itself is not a function of the snapshot, but I now realize that's absolutely wrong.